server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -352,7 +371,10 @@ public Map<String, Highlighter> getHighlighters() {
     }
 
     private ValuesSourceRegistry registerAggregations(List<SearchPlugin> plugins) {
-        ValuesSourceRegistry.Builder builder = new ValuesSourceRegistry.Builder();
+        ValuesSourceRegistry.Builder builder = new ValuesSourceRegistry.Builder(
```

Comment:
how about we move the registration closer to the usage - so inside the AggregationUsageService.Builder ?